### PR TITLE
codex-ui: colors

### LINKED ISSUE #93: UI themes

### PR DESCRIPTION
**Pull request by shvkhzod**
We created structure for storing themes and their corresponding colors. There will be different file for each theme in the folder themes, that file will have colors and their setup. 

Resolves #93

### CODE REVIEW

File: codex-ui/src/styles/themes/classic.pcss [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+body[theme-accent="classic"] {
+  --accent--bg-primary: var(--classic--bg-primary);
+  --accent--bg-secondary: var(--classic--bg-secondary);
+  --accent--bg-secondary-hover: var(--classic--bg-secondary-hover);
+  --accent-input: var(--classic--bg-input);
+  --accent--border: var(--classic--border);
+  --accent--solid: var(--classic--solid);
+  --accent-text-secondary: var(--classic--text-secondary);
+  --accent-text: var(--classic--text);
+  --accent-text-solid-foreground: var(--classic--text-solid-foreground);
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/classic.pcss:43]: --

File: codex-ui/src/styles/themes/crimson.pcss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+body[theme-accent="crimson"]{
+    /**
+    * Crimson Dark Colors
+    */
+    --crimson--bg-primary: #1D1A1B;
+    --crimson--bg-secondary: #272324;
+    --crimson--bg-secondary-hover:#322A2D;
+    --crimson--bg-input: #383134;
+    --crimson--border: #51383E;
+    --crimson--solid-secondary: # ;
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/crimson.pcss:12]: color missed

File: codex-ui/src/styles/themes/crimson.pcss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+*/
+body[theme-base="crimson"] {
+  --base--bg-primary: var(--crimson--bg-primary);
+  --base--bg-secondary: var(--crimson--bg-secondary);
+  --base--bg-secondary-hover: var(--crimson--bg-secondary-hover);
+  --base--bg-input: var(--crimson--bg-input);
+  --base--border: var(--crimson--border);
+  --base--solid-secondary: var(--crimson--solid-secondary);
+  --base--solid-hover: var(--crimson--solid-hover);
+  --base--solid: var( --crimson--solid);
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/crimson.pcss:31]:
```suggestion
  --base--solid: var(--crimson--solid);
```

File: codex-ui/src/styles/themes/grass.pcss [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+*/
+body[theme-base="grass"] {
+  --base--bg-primary: var(--grass--bg-primary);
+  --base--bg-secondary: var(--grass--bg-secondary);
+  --base--bg-secondary-hover: var(--grass--bg-secondary-hover);
+  --base--bg-input: var(--grass--bg-input);
+  --base--border: var(--grass--border);
+  --base--solid-secondary: var(--grass--solid-secondary);
+  --base--solid-hover: var(--grass--solid-hover);
+  --base--solid: var( --grass--solid);
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/grass.pcss:30]:
```suggestion
  --base--solid: var(--grass--solid);
```

File: codex-ui/src/styles/themes/red.pcss [hunk 5]
```diff
@@ -0,0 +1,49 @@
+
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/red.pcss:1]:
```suggestion
```

File: codex-ui/src/styles/themes/red.pcss [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+body[theme-base="red"],
+body[theme-accent="red"]{
+    /**
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/red.pcss:4]: use 2-space indentation, please

File: codex-ui/src/styles/themes/red.pcss [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+*/
+body[theme-base="red"] {
+  --base--bg-primary: var(--red--bg-primary);
+  --base--bg-secondary: var(--red--bg-secondary);
+  --base--bg-secondary-hover: var(--red--bg-secondary-hover);
+  --base--bg-input: var(--red--bg-input);
+  --base--border: var(--red--border);
+  --base--solid-secondary: var(--red--solid-secondary);
+  --base--solid-hover: var(--red--solid-hover);
+  --base--solid: var( --red--solid);
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/red.pcss:31]:
```suggestion
  --base--solid: var(--red--solid);
```

File: codex-ui/src/styles/themes/violet.pcss [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+*/
+body[theme-base="violet"] {
+  --base--bg-primary: var(--violet--bg-primary);
+  --base--bg-secondary: var(--violet--bg-secondary);
+  --base--bg-secondary-hover: var(--violet--bg-secondary-hover);
+  --base--bg-input: var(--violet--bg-input);
+  --base--border: var(--violet--border);
+  --base--solid-secondary: var(--violet--solid-secondary);
+  --base--solid-hover: var(--violet--solid-hover);
+  --base--solid: var( --violet--solid);
```
Comment (neSpecc) [inline codex-ui/src/styles/themes/violet.pcss:31]:
```suggestion
  --base--solid: var(--violet--solid);
```

File: codex-ui/src/vue/heading/Heading.vue [hunk 9]
```diff
@@ -19,7 +19,7 @@ const props = defineProps<{
 </script>
 
 <style module>
-@import '@/styles/mixins/typography.pcss';
+@import '../../styles/mixins/typography.pcss';
```
Comment (neSpecc) [inline codex-ui/src/vue/heading/Heading.vue:22]: why this has been changed?

File: src/presentation/pages/Home.vue [hunk 10]
```diff
@@ -8,15 +8,18 @@
...
+    />
     <Input text="Input text" />
   </div>
 </template>
 
 <script setup lang="ts">
 import { useHead } from 'unhead';
 import { useI18n } from 'vue-i18n';
-import { Button, Input, Heading } from 'codex-ui/vue';
+import { Button, Input, Heading } from '../../../codex-ui/src/vue';
```
Comment (neSpecc) [inline src/presentation/pages/Home.vue:22]: it should be used from workspace, like it was done before

Comment (shvkhzod) [reply src/presentation/pages/Home.vue:22]: image

Because I was receiving error

File: codex-ui/src/styles/themes/red.pcss
Comment (neSpecc) [inline codex-ui/src/styles/themes/red.pcss:1]: Please, lint this and other files. There are problems with indentations.

File: src/presentation/styles/themes/yellow.pcss
Comment (neSpecc) [inline src/presentation/styles/themes/yellow.pcss:1]: seems redundant

File: src/presentation/pages/Home.vue (hunk 10 above)
Comment (neSpecc) [reply src/presentation/pages/Home.vue]: If `yarn build` is working, that means that there is a problem with VSCode. TS config aliases used to let VScode understand that import. I'm not sure why it is not working in your case, need to research it.

Comment (neSpecc) [reply src/presentation/pages/Home.vue]: make sure you've run `yarn build` in `codex-ui`

**Review by neSpecc — APPROVED** [review]

File: codex-ui/src/styles/themes/classic.pcss [hunk 11]
```diff
@@ -0,0 +1,44 @@
+body[theme-base='classic'],
+body[theme-accent='classic'] {
+  /**
+  * Classic Dark Colors
+  */
+  --classic--bg-primary: #000000;
```
Comment (TatianaFomina) [inline codex-ui/src/styles/themes/classic.pcss:6]: why not `--classic-bg-primary`?

File: codex-ui/src/styles/themes/red.pcss [hunk 12]
```diff
@@ -0,0 +1,48 @@
+body[theme-base='red'],
+body[theme-accent='red'] {
+  /**
+    * Red Dark Colors
+    */
```
Comment (TatianaFomina) [inline codex-ui/src/styles/themes/red.pcss:5]:
```suggestion
  /**
   * Red Dark Colors
   */
```

File: codex-ui/src/styles/themes/red.pcss [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+  --red--solid-hover: #751f22;
+  --red--solid: #b72a2e;
+  --red--text-secondary: #856766;
+  --red--text: #f0dcdf;
+  --red--text-solid-foreground: #ffebeb;
+}
+
+/**
+* Set up colors for the theme Red
+*/
```
Comment (TatianaFomina) [inline codex-ui/src/styles/themes/red.pcss:21]:
```suggestion
/**
 * Set up colors for the theme Red
 */
```

File: codex-ui/src/styles/themes/violet.pcss [hunk 14]
```diff
@@ -0,0 +1,48 @@
+body[theme-base='violet'],
+body[theme-accent='violet'] {
+  /**
+    * Violet Dark Colors
+    */
```
Comment (TatianaFomina) [inline codex-ui/src/styles/themes/violet.pcss:5]:
```suggestion
  /**
   * Violet Dark Colors
   */
```

File: codex-ui/src/styles/themes/violet.pcss [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+  --violet--solid-hover: #461577;
+  --violet--solid: #913be6;
+  --violet--text-secondary: #a389bd;
+  --violet--text: #d9cee5;
+  --violet--text-solid-foreground: #f2f1f4;
+}
+
+/**
+* Set up colors for the theme Violet
+*/
```
Comment (TatianaFomina) [inline codex-ui/src/styles/themes/violet.pcss:21]:
```suggestion
/**
 * Set up colors for the theme Violet
 */
```

File: codex-ui/src/styles/themes/classic.pcss (hunk 11 above)
Comment (neSpecc) [reply codex-ui/src/styles/themes/classic.pcss:6]: to better separate theme name from variable name. This variable won't be used directly, `--base--bg-primary` or `--accent--bg-primary` will be used in components

Comment (TatianaFomina) [reply codex-ui/src/styles/themes/classic.pcss:6]: very error-prone decision, in my opinion

Comment (neSpecc) [reply codex-ui/src/styles/themes/classic.pcss:6]: its like a modifier, added to the beginning. `--base-text` and `--base--text` have different meaning

**Review by TatianaFomina — APPROVED** [review]

**Pull request merged**